### DISCUSSION TITLE
Help with activateBlock

### DISCUSSION THREAD

**Discussion by xViada** [PrismarineJS/mineflayer#2465] [Q&A]
Sorry , I'm so noob and I'm usin google translate. I have tried to write a bot thatcan activate a button in front of it, but I get the error TypeError: bot.blockAt is not a function. The code is this, could you help me to correct it?

```
var v = require('vec3');

var v1 = v(-300, 62, 1000);

bot.on('chat', (username, message) => {
  if (message === 'home') { 
    bot.chat("te escucho")
    bot.lookAt(v1)
    bot.activateBlock(button)
  }
})

button = bot.blockAt(v1)
```

**Comment by amoraschi** [comment]
You need to wait for the bot to spawn, put that code inside a spawn event:
```js
bot.once('spawn', () => {
  ...
})
```
(You can always join the Discord server for faster answers)
